File: 0654_MaximumBinaryTree/MaximumBinaryTree.cs
public class Solution {
    public TreeNode ConstructMaximumBinaryTree(int[] nums) {
        return ConstructMaximumBinaryTree(nums, 0, nums.Length-1);
    }
    
    private TreeNode ConstructMaximumBinaryTree(int[] nums, int l , int r)
    {
        if(l > r) return null;
        
        var index = FindMaxIndex(nums, l , r);
        var root = new TreeNode(nums[index]);
        root.left = ConstructMaximumBinaryTree(nums, l , index-1);
        root.right = ConstructMaximumBinaryTree(nums, index+1, r);
        return root;
    }
    
    private int FindMaxIndex(int[] nums, int l , int r)
    {
        var index = l;
        for(int i=l+1;i<=r;i++)
        {
            if(nums[i] > nums[index]) index = i;
        }
        
        return index;
    }
}

File: 0654_MaximumBinaryTree/MaximumBinaryTree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
 
class Solution(object):
    def constructMaximumBinaryTree(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        if not nums:
            return 
        if len(nums)==1:
            return TreeNode(nums[0])
        pivot = max(nums)
        idx = nums.index(pivot)
        node = TreeNode(pivot)
        node.left = self.constructMaximumBinaryTree(nums[:idx])
        node.right = self.constructMaximumBinaryTree(nums[idx+1:])
        return node